common/component/postgresql/awsIAM/aws.go

```diff
@@ -228,13 +74,30 @@ func InitAWSDatabase(ctx context.Context, config *pgxpool.Config, timeout time.D
 		// Manually reset auth token with aws and reset the config password using the new iam token
 		pwd, err := GetAccessToken(ctx, pgConfig, region, awsAccessKey, awsSecretKey)
 		if err != nil {
-			return fmt.Errorf("failed to refresh access token for iam authentication with postgresql: %v", err)
+			return fmt.Errorf("failed to refresh access token for iam authentication with PostgreSQL: %w", err)
 		}
 
 		pgConfig.Password = pwd
 		return nil
 	}
 
+	db, err := pgxpool.NewWithConfig(ctx, config)
+	if err != nil {
+		return fmt.Errorf("failed to get PostgreSQL config: %w", err)
+	}
+
+	// Create database and user with proper iam role if not using an already created iam user
+
+	err = databases.CreateDatabaseIfNeeded(ctx, timeout, connString, db)
+	if err != nil {
+		return fmt.Errorf("failed create AWS database if needed %w", err)
+	}
+
+	err = databases.CreateUserAndRoleIfNeeded(ctx, timeout, connString, db)
+	if err != nil {
+		return fmt.Errorf("failed create AWS user and grant role if needed %w", err)
+	}
+
 	// TODO: add clean up hooks on user/db?
```

Comment:
What's this TODO for?

Reply:
if we need to have clean up logic, but I suppose if the user is specifying a db and user, then clean up is outside of the realm of dapr?

Reply:
Cleanup for what things?